File: scratch/inspect_db.js
const https = require('https');

const options = {
  hostname: 'dnwsjvxitcndeqepovvo.supabase.co',
  path: '/rest/v1/',
  headers: {
    'apikey': 'sb_publishable_P3cAlUDSsQGQCJ2H9xrE3Q_BcICUlEj'
  }
};

https.get(options, (res) => {
  let body = '';
  res.on('data', (chunk) => body += chunk);
  res.on('end', () => {
    try {
      const data = JSON.parse(body);
      const definitions = data.definitions || {};
      for (const table of ['mindmaps', 'public_mindmaps']) {
        if (definitions[table]) {
          console.log(`Table: ${table}`);
          const properties = definitions[table].properties || {};
          const required = definitions[table].required || [];
          for (const [col, info] of Object.entries(properties)) {
            const reqStr = required.includes(col) ? ' (REQUIRED)' : '';
            console.log(`  - ${col}: ${info.type} / ${info.format}${reqStr}`);
          }
        } else {
          console.log(`Table ${table} not found. Available tables:`, Object.keys(definitions));
        }
      }
    } catch (e) {
      console.error('Error parsing response:', e.message);
      console.log('Body:', body);
    }
  });
}).on('error', (e) => {
  console.error('Request error:', e.message);
});

File: scratch/inspect_db.py
import urllib.request
import json
import os

url = "https://dnwsjvxitcndeqepovvo.supabase.co/rest/v1/"
headers = {
    "apikey": "sb_publishable_P3cAlUDSsQGQCJ2H9xrE3Q_BcICUlEj"
}

req = urllib.request.Request(url, headers=headers)
try:
    with urllib.request.urlopen(req) as response:
        data = json.loads(response.read().decode('utf-8'))
        
        # print definitions of mindmaps and public_mindmaps
        definitions = data.get("definitions", {})
        
        for table in ["mindmaps", "public_mindmaps"]:
            if table in definitions:
                print(f"Table: {table}")
                properties = definitions[table].get("properties", {})
                required = definitions[table].get("required", [])
                for col, info in properties.items():
                    req_str = " (REQUIRED)" if col in required else ""
                    print(f"  - {col}: {info.get('type')} / {info.get('format')}{req_str}")
            else:
                print(f"Table {table} not found in definitions. Available keys: {list(definitions.keys())}")
except Exception as e:
    print(f"Error fetching API spec: {e}")
